### DISCUSSION TITLE
Menus that received the visual update are inconsistent

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/microsoft-ui-xaml#7470] [General]
Some controls with similar menus received the new visual update, but results are sligthly inconsistent. I made a table with the differences I found:
### Vertical menus
| Control                           | Item height | Menu padding                                                                                   | Padding between items | Remarks                                                    |
|-----------------------------------|-------------|------------------------------------------------------------------------------------------------|-----------------------|------------------------------------------------------------|
| CommandBar.SecondaryCommands      | 32          | 4                                                                                              | 0                     |                                                            |
| CommandBarFlyout                  | 40          | 3                                                                                              | 0                     |                                                            |
| MenuFlyoutItem, MenuFlyoutSubItem | 35          | 4                                                                                              | 6                     | The hitbox extends 1px outside the hover visual            |
| RadioMenuFlyoutItem               | 35          | 3                                                                                              | 5                     |                                                            |
| ToggleMenuFlyoutItem              | 35          | 3                                                                                              | 4                     |                                                            |
| NavigationView                    | 36          | * Left/Compact mode: 4<br><br>* Hierarchical menu:<br>- left & right: 4,<br>- top & bottom: 10 | 4                     | Overflow/Hierarchical menus have the old style in Top mode |
| ComboBox                          | 33          | - left & right: 5,<br>- top & bottom: 6                                                        | 4                     |                                                            |

### Horizontal menus
| Control                 | Menu height | Item height | Menu padding | Padding between items |
|-------------------------|-------------|-------------|--------------|-----------------------|
| NavigationView Top mode | 48          | 47          | 0            | 0                     |
| MenuBar                 | 40          | 40          | 0            | 0                     |
| CommandBar              | 48          | 30          | 9            | 0                     |

Current as of v2.6.0-prerelease.210315002

Edit: added horizontal menus

**Comment by mdtauk** [comment]
I would suggest that someone should be assigned to audit all menu flyout controls to ensure consistency is attained, and maintained during the Visual Refresh process.

**Comment by StephenLPeters** [comment]
@gabrielconl Thanks for bringing these up! we are aware that there are some things which still need polish, this work is very much still in progress. We can report back when we think we are done, if you'd be willing to give it another close look :)

**Comment by ghost** [comment]
### Vertical menus
I suggest having the vertical menu items be 32px tall. It's the standard height for controls, with the compact height being 24px.

It could also be used for Left NavigationView. Currently items are 36px tall, but at the cost of 4px all vertical menus could be made consistent.

### Horizontal menus:
| Control \ Property | Menu height | Item height | Spacing                |
|--------------------|-------------|-------------|------------------------|
| MenuBar            | 40          | 32          | 8,4                    |
| TabView            | 40          | 32          | -                      |
| Top NavigationView | 48          | 48          | 0                      |
| CommandBar         | 48          | 36          | 6                      |
| InkToolbar         | 44          | 36          | 4                      |
| Proposed for all   | 40          | 32          | 4 (except for TabView) |

Current vs Proposed:
![current](https://user-images.githubusercontent.com/84711285/123516588-8688eb00-d6a5-11eb-927b-61eb2c0785da.png)
![proposed](https://user-images.githubusercontent.com/84711285/123516589-87ba1800-d6a5-11eb-8879-75d67e7de342.png)

The horizontal menus would look much better with the same height if they were used together, or when having multiple apps side by side. They would also be consistent with the vertical menus this way.

Top NavigationView items could have a visible backplate:
![top_navview](https://user-images.githubusercontent.com/84711285/123516604-9bfe1500-d6a5-11eb-8847-c9ea77a77f98.png)
